### PR TITLE
[csharp][httpclient] Introduce FileParameter as an abstraction

### PR DESCRIPTION
**Pull request by Blackclaws**
This introduces a new class FileParameter as an abstraction to use in our ApiClients. It contains an automatic implicit conversion for existing stream parameters so its backwards compatible to existing code. 

The reason for this is, that there are platforms that have file like parameters and we currently have no way to pass a filename into our api clients aside from FileStreams. However for Browser platforms there are no FileStreams there are IBrowserFiles that do not map. So we need another abstraction.

This could be further expanded to add content types to the files.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02)

### CODE REVIEW

**Comment by wing328** [comment]
cc @lucamazzanti as well (new C# technical committee member)

File: modules/openapi-generator/src/main/resources/csharp-netcore/libraries/httpclient/ApiClient.mustache [hunk 1]
```diff
@@ -214,6 +214,7 @@ namespace {{packageName}}.Client
             _httpClient = new HttpClient(_httpClientHandler, true);
             _disposeClient = true;
             _baseUrl = basePath;
+            #warning
```
Comment (lucamazzanti) [inline modules/openapi-generator/src/main/resources/csharp-netcore/libraries/httpclient/ApiClient.mustache:217]: probably to remove, this generates a c# warning directive without a message

**Review by lucamazzanti — APPROVED** [review]
I had the same needs, and I must override the client locally to avoid to clone and customize the apiclient.mustache file. After that PR, i will revert my customization and keep using it vanilla 100%.

A really useful feature to support also uploads from FileStreams but with the need to override the filename during upload.

It seems good, just watch out the warning i think you left.

File: modules/openapi-generator/src/main/resources/csharp-netcore/libraries/httpclient/ApiClient.mustache (hunk 1 above)
Comment (Blackclaws) [reply modules/openapi-generator/src/main/resources/csharp-netcore/libraries/httpclient/ApiClient.mustache:217]: Ah yes, that was an accident I'll remove it

**Pull request merged**